server/service/src/sensor/update.rs

```diff
@@ -31,9 +35,19 @@ pub fn update_sensor(
         .connection
         .transaction_sync(|connection| {
             let sensor_row = validate(connection, &ctx.store_id, &input)?;
-            let updated_sensor_row = generate(input, sensor_row);
+            let updated_sensor_row = generate(input.clone(), sensor_row.clone());
             SensorRowRepository::new(&connection).upsert_one(&updated_sensor_row)?;
 
+            if sensor_row.location_id != input.location_id {
```

Comment:
Might be better to move to generate? 🤷🏻

Reply:
Think its ok, looking at other code... Or why do you think it should be there? generate just generates the sensor_row.

Reply:
Hmm since it's logging a field change 🤔  we do the stock line field changes in generate

Reply:
Only see a `log_stock_changes()` in `service/src/stock_line/update.rs` this looks nice to me because it is a really clear separation between the insertion/generated and the logging...